added first draft of mia-platform v13 doc overview (#1313)
mia-platform/documentation: docs/release-notes/mia-platform-v13.mdx

---
id: v13-overview
title: Mia-Platform v13
sidebar_label: v13 Overview
image: "img/release-note-link-preview.png"
---

**Welcome to Mia-Platform v13!**

# Compose your Platform
With our latest release, we empower you to "Compose your platform," offering the freedom to compose your platform just the way you envision it. Whether you're forging new paths or refining existing strategies, Mia-Platform v13 is here to support you every step of the way, giving you the flexibility and control you need to craft a solution that fits you.
Let's take a closer look at all the new features included in Mia-Platform v13.

## Enhance Extensibility

> In today's fast-paced tech landscape, adaptability is key. That's why Mia-Platform v13 takes extensibility to the next level, making it easier than ever to integrate third-party tools and custom functionalities into your platform. We welcome **Platforge**,  our brand new Extensibility toolkit, in the Mia-Platform eco-system!

Reviewer: ```suggestion
In today's fast-paced tech landscape, adaptability is key. That's why Mia-Platform v13 takes extensibility to the next level, making it easier than ever to integrate third-party tools and custom functionalities into your platform. We welcome **Platforge**, our brand new Extensibility toolkit, in the Mia-Platform eco-system!
```

Extend the Console as you like.

From now on, you can seamlessly integrate external web pages within Mia-Platform Console using iFrames. 

This empowers Developers and Owners to enrich their experience by effortlessly incorporating tools, dashboards, or external content in our IDP, streamlining development workflows and tailoring the product to suits the business.  

INSERT AN IMAGE

To learn more, please refer to the [documentation]().

INSERT THE WORKING LINK

#### Automatize IDP via API
To Be clarified with Giulio

#### React to Console actions using Webhooks

The console events on which to configure webhooks are getting richer and richer, allowing you to react and build custom flows and logic each time the event occurs. 

With the introduction of user management events and the introduction of the history of each webhook to facilitate monitoring and debugging, users can now use webhooks to extend console functionality.

INSERT AN IMAGE

To learn more, please refer to the [documentation](/development_suite/webhooks-and-events/webhooks#event-history).

#### Configure once, deploy everywhere
TODO

#### Connect Backstage portal to Mia-Platform

> TODO

Reviewer: ```suggestion

On Backstage you will find a Mia-Platform plugin for integrating Mia-Platform components into Backstage components. It keeps your information synchronized effortlessly and gains insights into your company's domains, projects, deployment environments, and repository links, all while exploring their interrelations.
```

## Simplify Composability
Intro

### Benefits
<!--- da rivedere -->
The main benefits for the user can be summarized by the following:

- Democratizing access to the IDP's internal tools and utilities
- Simplifying the way inner sourcing is used.
- Enrich the management of IDP functionalities

### Features
Let's go in deep with the new Features:

#### Enrich Meta Data and list all your data assets: data catalog

> Data Catalog

Reviewer: ```suggestion

Let's welcome our first release of **Data Catalog**, designed to provide the data governance team and data analysts with a centralized location where they are collected all metadata concerning the data assets of an organization.

Set up our Data Catalog Agent, to connect to different external sources, such as Oracle Database, Salesforce, Postgres Database, MySQL Database, Microsoft SQL Server Database, and to extract all metadata existing on your organization System of Records. 

Through Mia-Platform Console, it is possible to:

- Keep metadata up to date with the external System of Records

- Navigate data models in a centralized place

- Enrich metadata with descriptions and GDPR regulation properties for each table field

- Export your data models in .csv format

- Make metadata usable via Data Catalog API from third-party systems and custom frontends
```

#### Seamless Runtime Management with Fast Data Control Plane

> Control Plane

Reviewer: ```suggestion

Mia-Platform Fast Data has been empowered with a Runtime Management solution, **Fast Data Control Plane**.  

This Fast Data suite component provides a general overview of the Fast Data pipelines released on the runtime environment, and allows to monitor and govern the execution of your Fast Data system.  

From its user-friendly web interface, pause and resume the consumption of data streams along Fast Data pipelines with just a few clicks to strategically allocate resources consumption and efficiently utilize them during peak demand processes.

### Benefits

Many are the benefits of adopting the Fast Data control plan in your Fast Data Projects:

- **Optimize Resource Allocation**, by strategically pausing and resuming the consumption of data streams along the Fast Data pipelines

- **Initial Load and Full Refresh processes management**, thanks to a user-friendly interface and without altering your workloads configuration deployed in runtime 
 
- **Performance Testing and Simulation**, to simulate different scenarios for performance testing by observing your system behavior under various conditions

- **Enhanced System Reliability**, allowing you to gracefully handle unexpected situations and/or system maintenance periods while minimizing system downtime.
```

#### Streamlined how you build your Saga, now No-Code

> No-code flow manager

Reviewer: ```suggestion

Let's welcome a new brand section: the Flow Manager Configurator!
Streamline your workflow with an intuitive graphical interface, designed to simplify the configuration of saga flows for Flow Manager services
Enjoy no-code editing and seamless service linking, all compatible with Flow Manager versions 2.0.0 and up.
Explore [our documentation](link) for more details.
```

Extend the Console as you like.

From now on, you can seamlessly integrate external web pages within Mia-Platform Console using iFrames. 

This empowers Developers and Owners to enrich their experience by effortlessly incorporating tools, dashboards, or external content in our IDP, streamlining development workflows and tailoring the product to suits the business.  

INSERT AN IMAGE

To learn more, please refer to the [documentation]().

INSERT THE WORKING LINK

#### Automatize IDP via API
To Be clarified with Giulio

#### React to Console actions using Webhooks

The console events on which to configure webhooks are getting richer and richer, allowing you to react and build custom flows and logic each time the event occurs. 

With the introduction of user management events and the introduction of the history of each webhook to facilitate monitoring and debugging, users can now use webhooks to extend console functionality.

INSERT AN IMAGE

To learn more, please refer to the [documentation](/development_suite/webhooks-and-events/webhooks#event-history).

#### Configure once, deploy everywhere
TODO

#### Connect Backstage portal to Mia-Platform

> On Backstage Marketplace you will find Mia-Platform plugin for integrating Mia-Platform components into Backstage components. It keeps your information synchronized effortlessly and gains insights into your company's domains, projects, deployment environments, and repository links, all while exploring their interrelations.

Reviewer: ```suggestion
On Backstage Marketplace you will find Mia-Platform plugin for integrating Mia-Platform components into Backstage components.  
It keeps your information synchronized effortlessly and gains insights into your company's domains, projects, deployment environments, and repository links, all while exploring their interrelations.
```

## Simplify Composability
Intro

### Benefits
<!--- da rivedere -->
The main benefits for the user can be summarized by the following:

- Democratizing access to the IDP's internal tools and utilities
- Simplifying the way inner sourcing is used.
- Enrich the management of IDP functionalities

### Features
Let's go in deep with the new Features:

> #### Enrich Meta Data and list all your data assets: data catalog

Reviewer: ```suggestion
#### Enrich Meta Data and list all your data assets: Data Catalog
```

Let's welcome our first release of **Data Catalog**, designed to provide the data governance team and data analysts with a centralized location where they are collected all metadata concerning the data assets of an organization.

> 

Reviewer: ```suggestion

ADD IMAGE

```

Set up our Data Catalog Agent, to connect to different external sources, such as Oracle Database, Salesforce, Postgres Database, MySQL Database, Microsoft SQL Server Database, and to extract all metadata existing on your organization System of Records. 

Through Mia-Platform Console, it is possible to:

- Keep metadata up to date with the external System of Records

- Navigate data models in a centralized place

- Enrich metadata with descriptions and GDPR regulation properties for each table field

- Export your data models in .csv format

- Make metadata usable via Data Catalog API from third-party systems and custom frontends

#### Seamless Runtime Management with Fast Data Control Plane

Mia-Platform Fast Data has been empowered with a Runtime Management solution, **Fast Data Control Plane**.  

> 

Reviewer: ```suggestion

ADD IMAGE

```

This Fast Data suite component provides a general overview of the Fast Data pipelines released on the runtime environment, and allows to monitor and govern the execution of your Fast Data system.  

From its user-friendly web interface, pause and resume the consumption of data streams along Fast Data pipelines with just a few clicks to strategically allocate resources consumption and efficiently utilize them during peak demand processes.

> ### Benefits

Reviewer: ```suggestion
##### Benefits
```

Many are the benefits of adopting the Fast Data control plan in your Fast Data Projects:

- **Optimize Resource Allocation**, by strategically pausing and resuming the consumption of data streams along the Fast Data pipelines

- **Initial Load and Full Refresh processes management**, thanks to a user-friendly interface and without altering your workloads configuration deployed in runtime 
 
- **Performance Testing and Simulation**, to simulate different scenarios for performance testing by observing your system behavior under various conditions

- **Enhanced System Reliability**, allowing you to gracefully handle unexpected situations and/or system maintenance periods while minimizing system downtime.

#### Streamlined how you build your Saga, now No-Code

> Let's welcome a new brand section: the Flow Manager Configurator!
> Streamline your workflow with an intuitive graphical interface, designed to simplify the configuration of saga flows for Flow Manager services
> Enjoy no-code editing and seamless service linking, all compatible with Flow Manager versions 2.0.0 and up.
> Explore [our documentation](link) for more details.

Reviewer: ```suggestion
Let's welcome a new brand section: the Flow Manager Configurator!  
ADD IMAGE 
Streamline your workflow with an intuitive graphical interface, designed to simplify the configuration of saga flows for Flow Manager services.  
Enjoy no-code editing and seamless service linking, all compatible with Flow Manager versions 2.0.0 and up. 
Explore [our documentation](link) for more details.
```

From its user-friendly web interface, pause and resume the consumption of data streams along Fast Data pipelines with just a few clicks to strategically allocate resources consumption and efficiently utilize them during peak demand processes.

##### Benefits

Many are the benefits of adopting the Fast Data control plan in your Fast Data Projects:

- **Optimize Resource Allocation**, by strategically pausing and resuming the consumption of data streams along the Fast Data pipelines

- **Initial Load and Full Refresh processes management**, thanks to a user-friendly interface and without altering your workloads configuration deployed in runtime 
 
- **Performance Testing and Simulation**, to simulate different scenarios for performance testing by observing your system behavior under various conditions

- **Enhanced System Reliability**, allowing you to gracefully handle unexpected situations and/or system maintenance periods while minimizing system downtime.

#### Streamlined how you build your Saga, now No-Code

> Let's welcome a new brand section: the Flow Manager Configurator!  
> ADD IMAGE 
> Streamline your workflow with an intuitive graphical interface, designed to simplify the configuration of saga flows for Flow Manager services.  
> Enjoy no-code editing and seamless service linking, all compatible with Flow Manager versions 2.0.0 and up. 
> Explore [our documentation](link) for more details.

Reviewer: ```suggestion
Let's welcome a new brand section: the Flow Manager Configurator!

ADD IMAGE  

Streamline your workflow with an intuitive graphical interface, designed to simplify the configuration of saga flows for Flow Manager services.  
Enjoy no-code editing and seamless service linking, all compatible with Flow Manager versions 2.0.0 and up.  
Explore [our documentation](link) for more details.
```

- Democratizing access to the IDP's internal tools and utilities
- Simplifying the way inner sourcing is used.
- Enrich the management of IDP functionalities

### Features
Let's go in deep with the new Features:

#### Enrich Meta Data and list all your data assets: Data Catalog

Let's welcome our first release of **Data Catalog**, designed to provide the data governance team and data analysts with a centralized location where they are collected all metadata concerning the data assets of an organization.

ADD IMAGE

Set up our Data Catalog Agent, to connect to different external sources, such as Oracle Database, Salesforce, Postgres Database, MySQL Database, Microsoft SQL Server Database, and to extract all metadata existing on your organization System of Records. 

Through Mia-Platform Console, it is possible to:

- Keep metadata up to date with the external System of Records

- Navigate data models in a centralized place

- Enrich metadata with descriptions and GDPR regulation properties for each table field

- Export your data models in .csv format

- Make metadata usable via Data Catalog API from third-party systems and custom frontends

> 

Reviewer: ```suggestion

To learn more, please refer to the [documentation](link).

```

ADD IMAGE

This Fast Data suite component provides a general overview of the Fast Data pipelines released on the runtime environment, and allows to monitor and govern the execution of your Fast Data system.  

From its user-friendly web interface, pause and resume the consumption of data streams along Fast Data pipelines with just a few clicks to strategically allocate resources consumption and efficiently utilize them during peak demand processes.

##### Benefits

Many are the benefits of adopting the Fast Data control plan in your Fast Data Projects:

- **Optimize Resource Allocation**, by strategically pausing and resuming the consumption of data streams along the Fast Data pipelines

- **Initial Load and Full Refresh processes management**, thanks to a user-friendly interface and without altering your workloads configuration deployed in runtime 
 
- **Performance Testing and Simulation**, to simulate different scenarios for performance testing by observing your system behavior under various conditions

- **Enhanced System Reliability**, allowing you to gracefully handle unexpected situations and/or system maintenance periods while minimizing system downtime.

> 

Reviewer: ```suggestion

To learn more, please refer to the [documentation](link).

```

INSERT AN IMAGE

To learn more, please refer to the [documentation]().

INSERT THE WORKING LINK

#### Automatize IDP via API
To Be clarified with Giulio

#### React to Console actions using Webhooks

The console events on which to configure webhooks are getting richer and richer, allowing you to react and build custom flows and logic each time the event occurs. 

With the introduction of user management events and the introduction of the history of each webhook to facilitate monitoring and debugging, users can now use webhooks to extend console functionality.

INSERT AN IMAGE

To learn more, please refer to the [documentation](/development_suite/webhooks-and-events/webhooks#event-history).

#### Configure once, deploy everywhere
TODO

#### Connect Backstage portal to Mia-Platform

On Backstage Marketplace you will find Mia-Platform plugin for integrating Mia-Platform components into Backstage components.  
It keeps your information synchronized effortlessly and gains insights into your company's domains, projects, deployment environments, and repository links, all while exploring their interrelations.

## Simplify Composability

> Intro

Reviewer: ```suggestion

Mia-Platform Suite empowers development teams to enhance software composability by facilitating streamlined reuse and embracing scalable approaches. With Mia-Platform v13, software composability is even more NoCode! 
```

### Benefits
<!--- da rivedere -->
The main benefits for the user can be summarized by the following:

- Democratizing access to the IDP's internal tools and utilities

> - Simplifying the way inner sourcing is used.

Reviewer: ```suggestion
- Providing user-friendly we interfaces for configuring, manage and monitoring your composable software
```

> - Enrich the management of IDP functionalities

Reviewer: ```suggestion
- Enriching the management of IDP with even more functionalities
```

sidebar_label: v13 Overview
image: "img/release-note-link-preview.png"
---

> :::info Mia-Platform Console v13.0.0 is now in Preview! and will be generally available on May 30th. :::

Reviewer: ```suggestion
:::info Mia-Platform Console v13.0.0 is **now in Preview** and will be generally available on May 30th. :::
```

**Welcome to Mia-Platform v13!**

# Compose your Platform
With our latest release, we empower you to "Compose your Platform," offering the freedom to compose your platform just the way you envision it. Whether you're forging new paths or refining existing strategies, Mia-Platform v13 is here to support you every step of the way, giving you the flexibility and control you need to craft a solution that fits you.
Let's take a closer look at all the new features included in Mia-Platform v13.

> ## Enhance Extensibility

Reviewer 2: ```suggestion
## Platforge: Enhance Extensibility
```

In today's fast-paced tech landscape, adaptability is key. That's why Mia-Platform v13 takes extensibility to the next level, making it easier than ever to integrate third-party tools and custom functionalities into your platform. We welcome **Platforge**, our brand new Extensibility toolkit, in the Mia-Platform eco-system!
The main benefits for the user can be summarized by the following:

- Extend developer lifecycle to be aligned with the needs of each team (Extend supported runtime, extend dev workflow, extend application management…).

> - Brownfield can standardize and simplify their developer workflow.

Reviewer 2: ```suggestion
- Brownfield users can standardize and simplify their developer workflow.
```

Reviewer 3: Are we sure that 'Brownfields' is a term familiar enough to our readers?

Author: yeah you may be right, we'll try to rephrase it

Let's go in deep with the new Features!

> ### Extend Frontend of Mia-Platform Console with iFrames

Reviewer 2: For now, you can make it only via API. Do We want to be explicit with this? Explaining that now you can use API, but we already planned that in the close future you can do it seamlessly.

Author: yes we may rephrase adding this information

Author: I added a proposal below

Extend the Console as you like.

From now on, you can seamlessly integrate external web pages within the Mia-Platform Console using iFrames. 

This empowers Console Users to enrich their experience by effortlessly incorporating tools, dashboards, or external content in our IDP, streamlining development workflows and tailoring the product to suit the business.  

INSERT AN IMAGE -> mettiamo immagine della sidebar con scritto my extension??????

To learn more, please refer to the [documentation]().

INSERT THE WORKING LINK

### React to Console actions using Webhooks

> The Console events on which to configure webhooks are getting richer and richer, allowing you to react and build custom flows and logic each time the event occurs. 

Reviewer 4: ```suggestion
The variety of Console events suitable for webhook configuration constantly increases, allowing you to react and build custom flows and logic each time the event occurs. 
```

With the introduction of user management events and the introduction of the history of each webhook to facilitate monitoring and debugging, users can now use webhooks to extend Console functionality.

INSERT AN IMAGE -> mettiamo la history dei webhook!

To learn more, please refer to the [documentation](/development_suite/webhooks-and-events/webhooks.md#event-history).

> ### Connect Backstage portal to Mia-Platform

Reviewer 2: If you want we can put a banner image that links to the Backstage Mkt or we can simply insert a link.

Author: this would be great!

Author: I don't understand what you mean :)

Author: @claudiafocchi aggiungiamo questa immagine. Capiamo se possiamo linkare in qualche modo la card di backstage sul loro sito. @Massibian il copy mi sembra corretto, @claudiafocchi ti torna?

On Backstage Marketplace you will find the Mia-Platform plugin for integrating Mia-Platform components into Backstage components.  

> It keeps your information synchronized effortlessly and gains insights into your Company's domains, Projects, deployment environments, and repository links, all while exploring their interrelations.

Reviewer 2: I think it is better to explicit the reason why this plugin is important.

Reviewer 3: Agree.
I would probably just write a line of context here: why should someone who uses React then want to use Mia

Author: @claudiafocchi can you pls try to add this information?

Let's welcome our first release of **Data Catalog**, designed to provide the data governance team and data analysts with a centralized location where all metadata concerning the data assets of an organization are collected.

ADD IMAGE -> dove facciamo vedere lo schemino architetturale

Set up our Data Catalog Agent, to connect to different external sources, such as Oracle Database, Salesforce, Postgres Database, MySQL Database, Microsoft SQL Server Database, and to extract all metadata existing on your organization's System of Records. 

Through Mia-Platform Console, it is possible to:

- Keep metadata up to date with the external System of Records
- Navigate data models in a centralized place
- Enrich metadata with descriptions and GDPR regulation properties for each table field
- Export your data models in .csv format
- Make metadata usable via Data Catalog API from third-party systems and custom frontends

To learn more, please refer to the [documentation](link).

### Seamless Runtime Management with Fast Data Control Plane

Mia-Platform Fast Data has been empowered with a Runtime Management solution, **Fast Data Control Plane**.  

> ADD IMAGE

Reviewer 2: We made a GIF 
![Gif fast data con border radius](https://github.com/mia-platform/documentation/assets/84580717/66bbc8f8-9564-4d02-b4ca-bf6ed25d77af)

Author: @albertotessarotto we can use this gif!

This Fast Data suite component provides a general overview of the Fast Data pipelines released on the runtime environment, and allows to monitor and govern the execution of your Fast Data system.  

> From its user-friendly web interface, pause and resume the consumption of data streams along Fast Data pipelines with just a few clicks to strategically allocate resources consumption and efficiently utilize them during peak demand processes.

Reviewer 4: ```suggestion
From its user-friendly web interface, you can pause and resume the consumption of data streams along Fast Data pipelines with just a few clicks to strategically allocate resources consumption and efficiently utilize them during peak demand processes.
```

Many are the benefits of adopting the Fast Data control plan in your Fast Data Projects, for example:

- **Optimize Resource Allocation**, by strategically pausing and resuming the consumption of data streams along the Fast Data pipelines
- **Initial Load and Full Refresh processes management**, thanks to a user-friendly interface and without altering the configuration of your workloads deployed in runtime 
- **Performance Testing and Simulation**, to simulate different scenarios for performance testing by observing your system behavior under various conditions
- **Enhanced System Reliability**, allowing you to gracefully handle unexpected situations and/or system maintenance periods while minimizing system downtime.

To learn more, please refer to the [documentation](link).

### Streamlined how you build your Saga, now No-Code

Let's welcome a new brand section: the Flow Manager Configurator!

> ADD IMAGE  

Reviewer 2: If you want we can embedd from YT a short video of rhea explaining the new flow manager

Author: this would be great! Do you have the link?

Reviewer 2: https://youtu.be/P6KesZaLNDM 

For now it is unlisted 'cuz we're missing description and subs. I ask to fix it as soon as possible.

ADD IMAGE

To learn more, please refer to the [documentation](link).

### Manage Sidecar containers with Mia-Platform Console

Governing Sidecar containers in your Projects has never been so simple: in the Design area of Mia-Platform Console, we have introduced a brand new `Sidecars` section where you can view and manage all the configurations of your Sidecar containers. You can also manage those Sidecar configurations in the detail page of each of your Microservices.
To discover more about how to configure and manage Sidecars, visit the related documentation page.

ADD LINK
ADD IMAGE

## Conversational DevX Powered by AI

This version of Mia-Platform Console is the first version powered by Artificial Intelligence! Welcome to our new Mia-Assistant that will support Mia-Platform users in speeding up onboarding and accelerating development.

The main benefits for the user can be summarized by the following:

- Speed up the onboarding when discovering our Mia-Platform Console
- Facilitates understanding of product features
- Simplify product discoverability
- Speed up the doc search
- Action Advisory
- Simplify access to features

Let's go in deep with the new Features:

> ### Mia-Assistant

Reviewer 2: Let's say that it is released in 'experiment' status, explaining what it means

Reviewer 5: I wrote in the last line that it is in "Experimental" phase. I think that it's quite self-explanatory. Or do you think that we should make it more explicit? In that case, have you got an alternative proposal?

**Welcome to Mia-Platform v13!**

# Compose your Platform

> With our latest release, we empower you to "Compose your Platform," offering the freedom to compose your platform just the way you envision it. Whether you're forging new paths or refining existing strategies, Mia-Platform v13 is here to support you every step of the way, giving you the flexibility and control you need to craft a solution that fits you.

Reviewer 3: ```suggestion
With our latest release, we empower you to "Compose your Platform", offering the freedom to compose your platform just the way you envision it. Whether you're forging new paths or refining existing strategies, Mia-Platform v13 is here to support you every step of the way, giving you the flexibility and control you need to craft a solution that fits you.
```

Let's take a closer look at all the new features included in Mia-Platform v13.

## Platforge: Enhance Extensibility

In today's fast-paced tech landscape, adaptability is key. That's why Mia-Platform v13 takes extensibility to the next level, making it easier than ever to integrate third-party tools and custom functionalities into your platform. We welcome **Platforge**, our brand new Extensibility toolkit, in the Mia-Platform eco-system!
The main benefits for the user can be summarized by the following:

- Extend developer lifecycle to be aligned with the needs of each team (Extend supported runtime, extend dev workflow, extend application management…).

> - Brownfield can standardize and simplify their developer workflow.

Author: ```suggestion
- Users that already use some specific tools can standardize and simplify their developer workflow, integrating them into Mia-Platform Console.
```

> - Users that already use some specific tools can standardize and simplify their developer workflow, integrating them into Mia-Platform Console.

Author: @akelity @Massibian what do you think about it? I removed Brownfield word

Let's go in deep with the new Features!

### Extend Frontend of Mia-Platform Console with iFrames

Extend the Console as you like.

From now on, you can seamlessly integrate external web pages within the Mia-Platform Console using iFrames. 

This empowers Console Users to enrich their experience by effortlessly incorporating tools, dashboards, or external content in our IDP, streamlining development workflows and tailoring the product to suit the business.  

> 

Author: ```suggestion
:::info
This feature is available only through API, so it is possible to leverage the Mia-Platform API Portal to insert the iFrame into a Console section. In future Console versions, there will be the possibility to do so through an User Interface.
:::
```

With the introduction of user management events and the introduction of the history of each webhook to facilitate monitoring and debugging, users can now use webhooks to extend Console functionality.

INSERT AN IMAGE -> mettiamo la history dei webhook!

To learn more, please refer to the [documentation](/development_suite/webhooks-and-events/webhooks.md#event-history).

### Connect Backstage portal to Mia-Platform

On Backstage Marketplace you will find the Mia-Platform plugin for integrating Mia-Platform components into Backstage components.  
It keeps your information synchronized effortlessly and gains insights into your Company's domains, Projects, deployment environments, and repository links, all while exploring their interrelations.

## Simplify Composability

Mia-Platform Suite empowers development teams to enhance software composability by facilitating streamlined reuse and embracing scalable approaches. With Mia-Platform v13, software composability is even more NoCode! 
The main benefits for the user can be summarized by the following:

> - Democratizing access to the IDP's internal tools and utilities

Reviewer 3: Here I might add that these tools are NO/LOW Code evolutions of their predecessors, which enhance the Developer Experience (rather than just being "user friendly")

> - Providing user-friendly interfaces to facilitate configuration, management and monitoring of your composable software

Author: ```suggestion
- Providing user-friendly interfaces to facilitate configuration, management ,and monitoring of your composable software
- Enhancing already existing No Code tools, taking the Developer Experience to the next level
```

ADD LINK
ADD IMAGE

## Conversational DevX Powered by AI

This version of Mia-Platform Console is the first version powered by Artificial Intelligence! Welcome to our new Mia-Assistant that will support Mia-Platform users in speeding up onboarding and accelerating development.

The main benefits for the user can be summarized by the following:

- Speed up the onboarding when discovering our Mia-Platform Console
- Facilitates understanding of product features
- Simplify product discoverability
- Speed up the doc search
- Action Advisory
- Simplify access to features

Let's go in deep with the new Features:

### Mia-Assistant

With this release we introduce a first version of our Mia-Assistant, an AI-based virtual chat that will make it easier for you to discover our Mia-Platform Console and access its documentation!
Mia-Assistant will answer your questions and provide you the related Mia-Platform documentation reference, to help you retrieve all the information you need and improve your experience with our product!
You can access Mia-Assistant anywhere in the Mia-Platform Console by simply clicking the "Ask Mia-Assistant" button in the bottom left-hand corner of any page. 

> This feature is now available in **Experimental** phase, so feel free to try it out and give us feedback in order to help us improve it! 

Reviewer 2: ```suggestion
This feature is currently in the Experimental phase.

:::info  
The 'Experimental' status suggests that while the feature is stable, the provided information might not be entirely accurate. We suggest caution in use
:::

We encourage you to give it a try and share your feedback with us. Your input will be invaluable in helping us refine and enhance it!"
```

Let's take a closer look at all the new features included in Mia-Platform v13.

## Platforge: Enhance Extensibility

In today's fast-paced tech landscape, adaptability is key. That's why Mia-Platform v13 takes extensibility to the next level, making it easier than ever to integrate third-party tools and custom functionalities into your platform. We welcome **Platforge**, our brand new Extensibility toolkit, in the Mia-Platform eco-system!
The main benefits for the user can be summarized by the following:

- Extend developer lifecycle to be aligned with the needs of each team (Extend supported runtime, extend dev workflow, extend application management…).
- Users that already use some specific tools can standardize and simplify their developer workflow, integrating them into Mia-Platform Console.

Let's go in deep with the new Features!

### Extend Frontend of Mia-Platform Console with iFrames

Extend the Console as you like.

> From now on, you can seamlessly integrate external web pages within the Mia-Platform Console using iFrames. 

Reviewer 4: ```suggestion
From now on, you can seamlessly integrate external web pages within Mia-Platform Console using iFrames. 
```

With the introduction of user management events and the introduction of the history of each webhook to facilitate monitoring and debugging, users can now use webhooks to extend Console functionality.

INSERT AN IMAGE -> mettiamo la history dei webhook!

To learn more, please refer to the [documentation](/development_suite/webhooks-and-events/webhooks.md#event-history).

### Connect Backstage portal to Mia-Platform

On Backstage Marketplace you will find the Mia-Platform plugin for integrating Mia-Platform components into Backstage components.  
It keeps your information synchronized effortlessly and gains insights into your Company's domains, Projects, deployment environments, and repository links, all while exploring their interrelations.

## Simplify Composability

Mia-Platform Suite empowers development teams to enhance software composability by facilitating streamlined reuse and embracing scalable approaches. With Mia-Platform v13, software composability is even more NoCode! 
The main benefits for the user can be summarized by the following:

- Democratizing access to the IDP's internal tools and utilities

> - Providing user-friendly interfaces to facilitate configuration, management ,and monitoring of your composable software

Reviewer 4: ```suggestion
- Providing user-friendly interfaces to facilitate configuration, management, and monitoring of your composable software
```

Through Mia-Platform Console, it is possible to:

- Keep metadata up to date with the external System of Records
- Navigate data models in a centralized place
- Enrich metadata with descriptions and GDPR regulation properties for each table field
- Export your data models in .csv format
- Make metadata usable via Data Catalog API from third-party systems and custom frontends

To learn more, please refer to the [documentation](link).

### Seamless Runtime Management with Fast Data Control Plane

Mia-Platform Fast Data has been empowered with a Runtime Management solution, **Fast Data Control Plane**.  

ADD IMAGE

This Fast Data suite component provides a general overview of the Fast Data pipelines released on the runtime environment, and allows to monitor and govern the execution of your Fast Data system.  
From its user-friendly web interface, you can pause and resume the consumption of data streams along Fast Data pipelines with just a few clicks to strategically allocate resources consumption and efficiently utilize them during peak demand processes.

> Many are the benefits of adopting the Fast Data control plan in your Fast Data Projects, for example:

Reviewer 4: ```suggestion
Many are the benefits of adopting the Fast Data control plane in your Fast Data Projects, for example:
```

Forget about using the CMS and use the Console as the star center to manage and govern the entire project lifecycle!

ADD IMAGE

To learn more, please refer to the [documentation](link).

### Manage Sidecar containers with Mia-Platform Console

Governing Sidecar containers in your Projects has never been so simple: in the Design area of Mia-Platform Console, we have introduced a brand new `Sidecars` section where you can view and manage all the configurations of your Sidecar containers. You can also manage those Sidecar configurations in the detail page of each of your Microservices.
To discover more about how to configure and manage Sidecars, visit the related documentation page.

ADD LINK
ADD IMAGE

## Conversational DevX Powered by AI

This version of Mia-Platform Console is the first version powered by Artificial Intelligence! Welcome to our new Mia-Assistant that will support Mia-Platform users in speeding up onboarding and accelerating development.

The main benefits for the user can be summarized by the following:

> - Speed up the onboarding when discovering our Mia-Platform Console

Reviewer 4: ```suggestion
- Speed up the onboarding when discovering Mia-Platform Console
```

> - Facilitates understanding of product features

Reviewer 4: ```suggestion
- Facilitate understanding of product features
```

- Simplify product discoverability
- Speed up the doc search
- Action Advisory
- Simplify access to features

Let's go in deep with the new Features:

### Mia-Assistant

With this release we introduce a first version of our Mia-Assistant, an AI-based virtual chat that will make it easier for you to discover our Mia-Platform Console and access its documentation!
Mia-Assistant will answer your questions and provide you the related Mia-Platform documentation reference, to help you retrieve all the information you need and improve your experience with our product!

> You can access Mia-Assistant anywhere in the Mia-Platform Console by simply clicking the "Ask Mia-Assistant" button in the bottom left-hand corner of any page. 

Reviewer 4: ```suggestion
You can access Mia-Assistant anywhere in Mia-Platform Console by simply clicking the "Ask Mia-Assistant" button in the bottom left-hand corner of any page. 
```

This function is now only accessible via Application Programming Interface (API), so it is possible to use the Mia-Platform API portal to integrate the iFrame into different sections of Mia-Platform Console. In future releases, we aim to simplify the integration of iFrame by providing directly a UI.

![iFrame Extensions](img/iframe-extensions.png)

To learn more, please refer to the [documentation]().

INSERT THE WORKING LINK

### React to Console actions using Webhooks

The variety of Console events suitable for webhook configuration constantly increases, allowing you to react and build custom flows and logic each time the event occurs. 

With the introduction of user management events and the introduction of the history of each webhook to facilitate monitoring and debugging, users can now use webhooks to extend Console functionality.

![Webhook History](img/event-history.png)

To learn more, please refer to the [documentation](/development_suite/webhooks-and-events/webhooks.md#event-history).

### Connect Backstage portal to Mia-Platform

> On Backstage Marketplace you will find the Mia-Platform plugin for integrating Mia-Platform components into Backstage components.  

Reviewer 6: ```suggestion
With the Mia-Platform Backstage Plugin is possible to kickstart your platform worry-free, starting from the components already on the Mia-Platform Console.
```

Let's welcome a new brand section: the Flow Manager Configurator!

ADD IMAGE  

Streamline your workflow with an intuitive graphical interface, designed to simplify the configuration of saga flows for Flow Manager services.  
Enjoy no-code editing and seamless service linking, all compatible with Flow Manager versions 2.0.0 and up.  
Explore [our documentation](/development_suite/api-console/api-design/flow-manager-configurator/overview.md) for more details.

### Blueprint and Template management via User Interface

Managing the company's Blueprint with their respective Templates and settings is now a child's play: with a dedicated interface, Templates, Environments, and Workload & Runtime settings for all new Company Projects can be created, set up, and modified in one click.

Forget about using the CMS and use the Console as the star center to manage and govern the entire project lifecycle!

![Project Blueprint](img/project-blueprint.png)

> To learn more, please refer to the related [documentation](/docs/development_suite/company/project-blueprint.md).

Reviewer: ```suggestion
To learn more, please refer to the related [documentation](/development_suite/company/project-blueprint.md).
```
